control/state.py

```diff
@@ -222,6 +227,53 @@ def get_omap_version(self) -> int:
                 f" invalid number of values ({value_list}).")
             raise
 
+    def lock_omap(self, timeout = None):
+        got_lock = False
+
+        if timeout is None:
+            self.logger.warning(f"Can't lock OMAP file without a lock timeout")
+            return False
+
+        for i in range(1, self.omap_file_lock_retries):
+            try:
+                with rados.WriteOpCtx() as write_op:
+                    self.ioctx.lock_exclusive(self.omap_name, self.OMAP_FILE_LOCK_NAME, self.OMAP_FILE_LOCK_COOKIE, "OMAP file changes lock", timeout, 0)
```

Comment:
I could be mistaken, but it seems that the "timeout" parameter in the "lock_exclusive()" function is essentially representing the "duration" of the lock. 

``` * @param duration the duration of the lock. Set to NULL for infinite duration```  see [here](https://github.com/ceph/ceph/blob/297ca51a5e3e023e54a9739c549165b3ff020ba2/src/include/rados/librados.h#L3637-L3655)

What if we pass "None" as the duration parameter? This raises the question of whether we still require user-defined lock timeout configurations or if they can be omitted altogether.

Reply:
@baum we can't pass None as a parameter. This parameter is read from the conf file and it must be an integer. The check for None remained from when the handling of the timeout default was done in state.py. Now it's done in a higher level. I left the check to make sure we didn't pass an invalid value. I removed it now and will push it later. I verified that the parsing function verifies the validity of the value for us.

Reply:
@gbregman do you agree that the parameter passed to lock_exclusive() as 'timeout' is actually 'duration'?

Reply:
Not that is matters much but I'll rename timeout to duration.